project-koku/koku
# [COST-1325] openshift on cloud cost deduplication

## Summary
This updates our OCP on Cloud SQL, specifically for the by OCP project view of the data to divide costs by duplicated row numbers to end up with the correct cost. 

- Revert the revert of processing changes back to calculating using the daily summary tables and "per project" instead of "per pod" 
- Divide cost by share projects and shared distinct pod labels in the project_daily_summary table to deduplicate cost
- Insert new line items into the ocp daily summary table for infrastructure raw cost instead of updating existing rows

## Testing


1. `git checkout cost-1325-openshift-on-cloud-cost-deduplication`
2. `make create-test-customer`
3. `make load-test-customer-data`
4. Call http://127.0.0.1:8000/api/cost-management/v1/reports/openshift/costs/?filter[time_scope_value]=-1&group_by[cluster]=*
5. Call http://127.0.0.1:8000/api/cost-management/v1/reports/openshift/infrastructures/aws/costs/?filter[time_scope_value]=-1
6. Call http://127.0.0.1:8000/api/cost-management/v1/reports/openshift/infrastructures/azure/costs/?filter[time_scope_value]=-1
7. Confirm that OCP on AWS raw cost + markup is equal to the raw infrastructure cost for my-ocp-cluster-1 on the openshift cost API
8. Confirm that OCP on Azure raw cost + markup is equal to the raw infrastructure cost for my-ocp-cluster-2 on the openshift cost API 
9. `git checkout master`
10. Call http://127.0.0.1:5000/api/cost-management/v1/report_data/?provider_uuid=56292c6e-b8b0-4533-8eb0-0e4a05d30fbc&schema=acct10001&start_date=2021-05-01 for each OCP source, substituting the the correct uuid
11. Refresh the reports APIs and see that the costs no longer add up
